error when install Realm on window

Hi all,

I have try to install Realm on window for new project react-native, but it fall

step I did: 

create new react-native project
 `react-native init TestRealm`
`cd TestRealm`

And then install Realm

`npm install --save realm`

And link

`react-native link realm`

Then run 

`react-native run-android`

but it have error

```
C:\Users\keyyu\Atom\Workspace\TestRealm>react-native run-android
Starting JS server...
'adb' is not recognized as an internal or external command,
operable program or batch file.
Building and installing the app on the device (cd android && gradlew.bat installDebug)...

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\keyyu\Atom\Workspace\TestRealm\node_modules\realm\android\build.gradle' line: 71

* What went wrong:
A problem occurred evaluating project ':realm'.
> Cannot run program "npm" (in directory "C:\Users\keyyu\Atom\Workspace\TestRealm\node_modules\realm\android"): CreateProcess error=2, The system cannot find the file specified

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 22.896 secs
Could not install the app on the device, read the error above for details.
Make sure you have an Android emulator running or a device connected and have
set up your Android development environment:
https://facebook.github.io/react-native/docs/android-setup.html

(node:10272) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: Cannot read property 'message' of undefined
```


I did step by step like [https://realm.io/docs/react-native/latest/](https://realm.io/docs/react-native/latest/).

my enviroment

"node": v6.9.2
"npm":  v3.10.9
"react": "15.4.2",
"react-native": "0.40.0",
"realm": "^0.15.4"


Please help. 

Thank you!

I did run into same issue, but after commenting out the `version` in `..\node_modules\realm\android\build.gradle` another error popped up.
```FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':realm:compileReleaseJavaWithJavac'.
> Could not find tools.jar. Please check that C:\Program Files\Java\jre1.8.0_121 contains a valid JDK installation.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.```
But only when trying to run via console. Seems to work fine with commented out `version` while building via android studio.

--- a/react-native/android/publish_android_template
+++ b/react-native/android/publish_android_template
@@ -66,9 +66,17 @@ String getAppId () {
 
 apply from: 'analytics.gradle'
 
+import groovy.json.JsonSlurper
+
+def getNpmVersion() {
+    def inputFile = new File("../package.json")
+    def packageJson = new JsonSlurper().parseText(inputFile.text)
+    return packageJson["version"]
+}
+
 task send(type: SendAnalyticsTask) {
     applicationId = getAppId()
-    version = "npm --silent run get-version".execute(null, projectDir).text.trim()
+    version = getNpmVersion().trim()
 }
 
 preBuild.dependsOn send

--- a/react-native/android/publish_android_template
+++ b/react-native/android/publish_android_template
@@ -69,7 +69,7 @@ apply from: 'analytics.gradle'
 import groovy.json.JsonSlurper
 
 def getNpmVersion() {
-    def inputFile = new File("../package.json")
+    def inputFile = new File(buildscript.sourceFile.getParent() + "/../package.json")
     def packageJson = new JsonSlurper().parseText(inputFile.text)
     return packageJson["version"]
 }

